### PR TITLE
Cleaning up compatibility

### PR DESCRIPTION
**Pull request by namannimmo10**
#### References to other Issues or PRs

#18286   #17943 

#### Brief description of what is fixed or changed
3. Removed `xrange`. 
4. Removed `range` import from compatibility in various files. 

#### Release Notes

NO ENTRY

### CODE REVIEW

**Comment by ShubhamKJha** [comment]
I believe you've already removed `string_types` from codebase. 
You should pull the recent master first (which has your wok merged), if you have not already done so.
I would suggest to first remove all the imports of `range` from `core.compatibility`. A simple git grep can give some insights.
```bash
git grep -l 'compatibiliy import .* range'    # lists almost all the files that import range

git grep -l  ".*range\(,\|)\)"      # should list the remaining files (if any)
```
Then remove it from `compatibility`.

**Comment by czgdp1807** [comment]
I think `master` should be merged into `remove-py2`.

**Comment by namannimmo10** [comment]
Thanks a lot, @ShubhamKJha @czgdp1807. :) It's ready for review!

File: sympy/sets/fancysets.py [hunk 1]
```diff
@@ -3,7 +3,7 @@
 from functools import reduce
 
 from sympy.core.basic import Basic
-from sympy.core.compatibility import with_metaclass, range, PY3
```
Comment (ShubhamKJha) [inline sympy/sets/fancysets.py]: Is removing `PY3` intended?

Comment (namannimmo10) [reply sympy/sets/fancysets.py]: Unused import.

**Comment by ShubhamKJha** [comment]
Looks good till now. You can remove `long` and `round` imports then we can merge it.

**Comment by namannimmo10** [comment]
Okay, I'll update soon. Just let me know if something else is also required.

**Comment by namannimmo10** [comment]
> You can remove `long` and `round` imports

This piece of work is a little tricky as of now. Bunch of failing tests....

**Comment by namannimmo10** [comment]
@ShubhamKJha, please merge this.

**Pull request merged**